File: src/nodejs/hprose/hprose.js
/**********************************************************\
|                                                          |
|                          hprose                          |
|                                                          |
| Official WebSite: http://www.hprose.com/                 |
|                   http://www.hprose.net/                 |
|                   http://www.hprose.org/                 |
|                                                          |
\**********************************************************/

/**********************************************************\
 *                                                        *
 * hprose.js                                              *
 *                                                        *
 * hprose for Node.js.                                    *
 *                                                        *
 * LastModified: Nov 18, 2013                             *
 * Author: Ma Bingyao <andot@hprfc.com>                   *
 *                                                        *
\**********************************************************/

global.HproseException = require('./common/HproseException.js');
global.HproseResultMode = require('./common/HproseResultMode.js');
global.HproseFilter = require('./common/HproseFilter.js');
global.HproseClassManager = require('./io/HproseClassManager.js'),
global.HproseBufferInputStream = require('./io/HproseBufferInputStream.js');
global.HproseBufferOutputStream = require('./io/HproseBufferOutputStream.js');
global.HproseTags = require('./io/HproseTags.js');
global.HproseRawReader = require('./io/HproseRawReader.js');
global.HproseSimpleReader = require('./io/HproseSimpleReader.js');
global.HproseSimpleWriter = require('./io/HproseSimpleWriter.js');
global.HproseReader = require('./io/HproseReader.js');
global.HproseWriter = require('./io/HproseWriter.js');
global.HproseFormatter = require('./io/HproseFormatter.js');
global.HproseService = require('./server/HproseService.js');
global.HproseHttpService = require('./server/HproseHttpService.js');
global.HproseHttpServer = require('./server/HproseHttpServer.js');
global.HproseClient = require('./client/HproseClient.js');
global.HproseHttpClient = require('./client/HproseHttpClient.js');

module.exports = {
    common: {
        Exception: global.HproseException,
        ResultMode: global.HproseResultMode,
        Filter: global.HproseFilter
    },
    io: {
        ClassManager: global.HproseClassManager,
        BufferInputStream: global.HproseBufferInputStream,
        BufferOutputStream: global.HproseBufferOutputStream,
        Tags: global.HproseTags,
        RawReader: global.HproseRawReader,
        SimpleReader: global.HproseSimpleReader,
        SimpleWriter: global.HproseSimpleWriter,
        Reader: global.HproseReader,
        Writer: global.HproseWriter,
        Formatter: global.HproseFormatter
    },
    server: {
        Service: global.HproseService,
        HttpService: global.HproseHttpService,
        HttpServer: global.HproseHttpServer
    },
    client: {
        Client: global.HproseClient,
        HttpClient: global.HproseHttpClient
    }
}

File: src/ruby/lib/hprose.rb
############################################################
#                                                          #
#                          hprose                          #
#                                                          #
# Official WebSite: http://www.hprose.com/                 #
#                   http://www.hprose.net/                 #
#                   http://www.hprose.org/                 #
#                                                          #
############################################################

############################################################
#                                                          #
# hprose.rb                                                #
#                                                          #
# hprose for ruby                                          #
#                                                          #
# LastModified: Dec 1, 2012                                #
# Author: Ma Bingyao <andot@hprfc.com>                     #
#                                                          #
############################################################

module Hprose
  autoload :Exception, 'hprose/common'
  autoload :ResultMode, 'hprose/common'
  autoload :Filter, 'hprose/common'
  autoload :Tags, 'hprose/io'
  autoload :ClassManager, 'hprose/io'
  autoload :Reader, 'hprose/io'
  autoload :Writer, 'hprose/io'
  autoload :Formatter, 'hprose/io'
  autoload :Client, 'hprose/client'
  autoload :HttpClient, 'hprose/httpclient'
  autoload :Service, 'hprose/service'
  autoload :HttpService, 'hprose/httpservice'
end

Object.const_set(:HproseException, Hprose.const_get(:Exception))
Object.const_set(:HproseResultMode, Hprose.const_get(:ResultMode))
Object.const_set(:HproseFilter, Hprose.const_get(:Filter))
Object.const_set(:HproseTags, Hprose.const_get(:Tags))
Object.const_set(:HproseClassManager, Hprose.const_get(:ClassManager))
Object.const_set(:HproseReader, Hprose.const_get(:Reader))
Object.const_set(:HproseWriter, Hprose.const_get(:Writer))
Object.const_set(:HproseFormatter, Hprose.const_get(:Formatter))
Object.const_set(:HproseClient, Hprose.const_get(:Client))
Object.const_set(:HproseHttpClient, Hprose.const_get(:HttpClient))
Object.const_set(:HproseService, Hprose.const_get(:Service))
Object.const_set(:HproseHttpService, Hprose.const_get(:HttpService))